cpp/ps0005/p0005.cpp

```diff
@@ -27,24 +30,31 @@ int main() {
                 result = num1 / num2;
             } else {
                 cout << "Error: Division by zero" ;
-                return 1; // Exit the program
-            }
+                return 2; // Exit the program
+            } // 1/1
             break;
         case '%':
             if (int(num2)!= 0) {
-                result = int(num1) % int(num2);
+                if (int(num1) == num1 && int(num2) == num2))
+                    result = int(num1) % int(num2);
+                else
+                {
+                    cout << "cannot execute modulo operation for floating point numbers." << endl;
+                    return 4;
+                }
             } else {
                 cout << "Error: Division by zero" ;
-                return 1; 
-            }
+                return 2; 
+            } // 0.5/1
             break;
         default:
             cout << "Invalid operation!"  ;
-            return 1; 
-    }
+            return 1; // 1/1 
+    } // 0.5/1
 
-   
+    // printing the result if all data is valid 1/1
     cout << "Result: " << result << endl;
 
     return 0;
 }
+// great work 😎 8/10 ⭐

```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [x] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [ ] commented some phrases for blocks of code
- [x] main returns non-zero/zero number according to the case

you can check the [problem solution](https://onlinegdb.com/2mF7i4R8H) for more